pkg/skaffold/log/log.go

```diff
@@ -14,24 +14,40 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1
+package log
 
 import (
+	"context"
 	"io"
+	"time"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/graph"
-	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/logger"
 )
 
-func (r *SkaffoldRunner) createLogger(out io.Writer, artifacts []graph.Artifact) *logger.LogAggregator {
-	if !r.runCtx.Tail() {
-		return nil
-	}
+type Logger interface {
+	StartLogger(context.Context, io.Writer, []string) error
```

Comment:
I actually did this intentionally - since these methods are now embedded in the `Deployer`, we'd be calling `Deployer.Start()` to start the `Logger`. I didn't like this as it's actually less clear, so to avoid ambiguity I made them specific to the `Logger`.

Reply:
Ah ok.  I forgot its embedded interface.